doc/modules/feature_selection.rst

```diff
@@ -234,11 +234,27 @@ Randomized sparse models
 
 .. currentmodule:: sklearn.linear_model
 
-The limitation of L1-based sparse models is that faced with a group of
-very correlated features, they will select only one. To mitigate this
-problem, it is possible to use randomization techniques, reestimating the
-sparse model many times perturbing the design matrix or sub-sampling data
-and counting how many times a given regressor is selected.
+In terms of feature selection, there are some well-known limitations of
+L1-penalized models for regression and classification.  For example, it is
+known that the Lasso will tend to select an individual variable out of a group
+of highly correlated features.  Furthermore, even when the correlation between
+features is not too high, the conditions under which L1-penalized methods
+consistently select "good" features can be restrictive in general.
+
+To mitigate this problem, it is possible to use a randomization technique
+known as stability selection.  In this method, a subsample of the data is fit
+to a L1-penalized model where the penalty of a random subset of coefficients
+has been scaled.  Specifically, given a subsample of the data :math:`(y_i,
+x_i), i \in I`, where :math:`I \subset \{1, 2, \ldots, n\}` is a random subset
+of the data of size :math:`n_I`, the following modified Lasso fit is obtained:
+
+.. math::   \hat{w_I} = \mathrm{arg}\min_{w} \frac{1}{2n_I} \sum_{i \in I} (y_i - x_i^T w)^2 + \alpha \sum_{j=1}^p \frac{|w_j|}{s_j},
+
+where :math:`s_j \in \{s, 1\}` are independent trials of a fair Bernoulli
```

Comment:
Very good observation. Though, to stay true with the source code, I would say it's better to replace `alpha` with `C` and `s` with `alpha`. We would ideally want someone to look at the model's description, and understand the model by looking at its objective function.

But you do make a good point about trying to stay true to the paper; I would put in parenthesis what the variables `C` and `alpha` are representing from the original paper. 

(Making the point that `w` here is `beta` in the paper, and `s_j` here is `w_k` in the paper seems like overkill.)